### PR TITLE
Enhance documentation

### PR DESCRIPTION
**Pull request by LeStegii**

### CODE REVIEW

File: minimessage/src/main/java/io/github/almightysatan/slams/minimessage/MMStringEntry.java [hunk 1]
```diff
@@ -29,8 +29,21 @@
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 
+/**
+ * {@link MMStringEntry}s represent a {@link String} in a language file parsed in MiniMessage format.
```
Comment (Almighty-Satan) [inline minimessage/src/main/java/io/github/almightysatan/slams/minimessage/MMStringEntry.java:33]:
```suggestion
 * Represents a message parsed in MiniMessage format.
```
I also really don't like the naming of this class. We should probably change it.

**Review by Almighty-Satan — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -1,20 +1,33 @@
...
 ```java
-LanguageManager langManager = LanguageManager.create("English"); // Set English as the default language
-langManager.load("English", JacksonParser.createJsonParser("messages.json")); // Register language and load messages from JSON file
+Slams slams = Slams.create("English"); // Set English as the default language
+slams.load("English", JacksonParser.createJsonParser("messages.json")); // Register language and load messages from JSON file
 
-MMStringEntry test0 = MMStringEntry.of("test0", langManager, null); // Just a simple message
-MMStringEntry test1 = MMStringEntry.of("test1", langManager, ctx -> Placeholder.unparsed("hello", "world")); // Message with placeholder, "hello" will be replaced with "world"
+AdventureMessage test0 = AdventureMessage.of("test0", slams, null); // Just a simple message
+AdventureMessage test1 = AdventureMessage.of("test1", slams, ctx -> Placeholder.unparsed("hello", "world")); // Message with placeholder, "hello" will be replaced with "world"
```
Comment (Almighty-Satan) [inline README.md:23]:
```suggestion
AdventureMessage test0 = AdventureMessage.of("test0", slams); // Just a simple message
AdventureMessage test1 = AdventureMessage.of("test1", slams, ctx -> net.kyori.adventure.text.minimessage.tag.resolver.Placeholder.unparsed("hello", "world")); // Message with placeholder, "hello" will be replaced with "world"

slams.load("English", JacksonParser.createJsonParser("messages.json")); // Register language and load messages from JSON file
```
Messages have to be registered before loading the language

File: README.md [hunk 3]
```diff
@@ -1,20 +1,33 @@
...
 
 player.sendMessage(test0.value(null)); // Send the message to a player. No context is provided and therefore the default language will be used. See Context#language
 player.sendMessage(test1.value(null, Placeholder.unparsed("123", "456"))); // Send another message but add an additional placeholder
```
Comment (Almighty-Satan) [inline README.md:26]:
```suggestion
audience.sendMessage(test0.value(null)); // Send the message to an Adventure Audience. No context is provided and therefore the default language will be used. See Context#language
audience.sendMessage(test1.value(null, net.kyori.adventure.text.minimessage.tag.resolver.Placeholder.unparsed("123", "456"))); // Send another message but add an additional placeholder
```

File: README.md [hunk 4]
```diff
@@ -1,20 +1,33 @@
...
 # SLAMS - Simple Language And Message System
 
+___
+
+SLAMS is an easy to use message/language system allowing projects to easily support multiple languages with placeholders and message formatting. Messages can be automatically loaded from formats like JSON or supplied by the user via a custom implementation of the `LanguageParser` interface.
+
+SLAMS has been created with support for Minecraft servers in mind and therefore also supports formats like [MiniMessages](https://docs.advntr.dev/minimessage/index.html).
+
 ### Standalone Example
 // TODO
```
Comment (Almighty-Satan) [inline README.md:15]:
```suggestion
Slams slams = Slams.create("English"); // Set English as the default language

StandaloneMessage test0 = StandaloneMessage.of("test0", slams); // Just a simple message
StandaloneMessage test1 = StandaloneMessage.of("test1", slams, Placeholder.constant("hello", "world")); // Message with placeholder, "hello" will be replaced with "world"

slams.load("English", JacksonParser.createJsonParser("messages.json")); // Register language and load messages from JSON file

System.out.println(test0.value()); // Print the message. No context is provided and therefore the default language will be used. See Context#language
System.out.println(test1.value(null, Placeholder.constant("123", "456"))); // Print another message but add an additional placeholder
```

File: README.md (hunk 3 above)
Comment (LeStegii) [reply README.md:36]: Added note about the type of placeholder

File: README.md (hunk 2 above)
Comment (LeStegii) [reply README.md]: Fixed in https://github.com/Almighty-Satan/SLAMS/pull/1/commits/5cf21cccc544f367c867c1d3be642d370f90c804

File: README.md [hunk 5]
```diff
@@ -1,20 +1,44 @@
...
 
 ### Mini-Message Example
 ```java
-LanguageManager langManager = LanguageManager.create("English"); // Set English as the default language
-langManager.load("English", JacksonParser.createJsonParser("messages.json")); // Register language and load messages from JSON file
+Slams slams = Slams.create("English"); // Set English as the default language
 
-MMStringEntry test0 = MMStringEntry.of("test0", langManager, null); // Just a simple message
-MMStringEntry test1 = MMStringEntry.of("test1", langManager, ctx -> Placeholder.unparsed("hello", "world")); // Message with placeholder, "hello" will be replaced with "world"
+AdventureMessage test0 = AdventureMessage.of("test0", slams, null); // Just a simple message
```
Comment (Almighty-Satan) [inline README.md:31]:
```suggestion
AdventureMessage test0 = AdventureMessage.of("test0", slams); // Just a simple message
```

File: README.md [hunk 6]
```diff
@@ -1,20 +1,44 @@
...
+AdventureMessage test1 = AdventureMessage.of("test1", slams, ctx -> Placeholder.unparsed("hello", "world")); // Message with placeholder, "hello" will be replaced with "world"
```
Comment (Almighty-Satan) [inline README.md:32]:
```suggestion
AdventureMessage test1 = AdventureMessage.of("test1", slams, ctx -> net.kyori.adventure.text.minimessage.tag.resolver.Placeholder.unparsed("hello", "world")); // Message with placeholder, "hello" will be replaced with "world"
```

File: README.md [hunk 7]
```diff
@@ -1,20 +1,44 @@
...
 
-player.sendMessage(test0.value(null)); // Send the message to a player. No context is provided and therefore the default language will be used. See Context#language
-player.sendMessage(test1.value(null, Placeholder.unparsed("123", "456"))); // Send another message but add an additional placeholder
+slams.load("English", JacksonParser.createJsonParser("messages.json")); // Register language and load messages from JSON file
+
+audience.sendMessage(test0.value(null)); // Send the message to an Adventure Audience. No context is provided and therefore the default language will be used. See Context#language
```
Comment (Almighty-Satan) [inline README.md:36]:
```suggestion
audience.sendMessage(test0.value()); // Send the message to an Adventure Audience. No context is provided and therefore the default language will be used. See Context#language
```

File: core/src/main/java/io/github/almightysatan/slams/Message.java [hunk 8]
```diff
@@ -25,21 +25,58 @@
...
+ * Represents a message. The value of a message is not necessarily a {@link String}. It could be a multidimensional
+ * array of Strings, some sort of Map, a MiniMessage Component or something completely different.
+ *
+ * @param <T> the type of this messages value
+ */
 public interface Message<T> {
 
+    /**
+     * The dotted path of this message.
+     *
```
Comment (LeStegii) [inline core/src/main/java/io/github/almightysatan/slams/Message.java:38]:
```suggestion
    /**
     * The dotted path of this message. For example 'path.to.example.message'.
     *
```

**Pull request merged**